File: packages/wasm-tools/examples/wiki.js
import createPlugin from "@extism/extism";

async function main() {
	const plugin = await createPlugin("../extism/wiki/wiki.wasm", {
		useWasi: true,
		functions: {},
		runInWorker: true,
		allowedHosts: ["*.wikipedia.org"],
		memory: { maxHttpResponseBytes: 100 * 1024 * 1024 },
	});

	try {
		const params = { query: "Ho Chi Minh City" };
		const data = await plugin.call("wikiCall", JSON.stringify(params));
		console.log(data.json()); 
	} catch (e) {
		console.error(e);
	} finally {
		await plugin.close();
	}
}

void main();

File: packages/wasm-tools/examples/wiki.py
import extism
import json
from os.path import join, dirname


def read_local_wasm(file_name):
    path = join(dirname(__file__), file_name)
    with open(path, "rb") as wasm_file:
        return wasm_file.read()


def _manifest(file_name):
    wasm = read_local_wasm(file_name)
    return {
        "wasm": [{"data": wasm}],
        "allowed_hosts": ["*.wikipedia.org"],
    }


manifest = _manifest("wiki.wasm")
with extism.Plugin(manifest, wasi=True) as plugin:
    data = plugin.call(
        "wikiCall",
        json.dumps({"query": "Ho Chi Minh City"}),
        parse=lambda output: json.loads(bytes(output).decode("utf-8")),
    )
print(data)
